persistence/sql/persister_identity.go

```diff
@@ -472,3 +512,110 @@ func (p *Persister) injectTraitsSchemaURL(ctx context.Context, i *identity.Ident
 	i.SchemaURL = s.SchemaURL(p.r.Config(ctx).SelfPublicURL(nil)).String()
 	return nil
 }
+
+func (p *Persister) getJsonSearchQuery(field string, values []string) pop.ScopeFunc {
+	return func(q *pop.Query) *pop.Query {
+		switch p.Connection(context.Background()).Dialect.Name() {
+		case "sqlite3", "mysql", "mariadb":
+			field, innerField := extractFieldAndInnerFields(field)
+			for _, value := range values {
+				if innerField == "" {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$') = ?`, field), value)
```

Comment:
This is vulnerable to SQL Injection. I saw that there is a sanitize function further down but I do not trust it and in the future developers might mess up the use. Best would be to allow only regex `[a-zA-Z0-9\.]+` here where \. is the delimiter and directly sanitize the field in the Sprintf statement.

Reply:
and `_`